src/rest_vol_dataset.c

```diff
@@ -3991,15 +4039,29 @@ RV_convert_dataspace_selection_to_string(hid_t space_id, char **selection_string
                 break;
 
             case H5S_SEL_POINTS:
-                FUNC_GOTO_ERROR(H5E_DATASPACE, H5E_UNSUPPORTED, FAIL,
-                                "point selections are unsupported as a HTTP request parameter");
+                hssize_t num_points;
+                size_t   point_buf_size;
+#ifdef RV_CONNECTOR_DEBUG
+                printf("-> Point selection\n\n");
+#endif
+                /* Format the point selection as a binary buffer containing each coordinate index as an
+                 * hsize_t */
+                if ((num_points = H5Sget_select_npoints(space_id)) < 0)
+                    FUNC_GOTO_ERROR(H5E_DATASPACE, H5E_CANTGET, FAIL, "can't get number of selected points");
+
+                point_buf_size = (size_t)(ndims * num_points) * sizeof(hsize_t);
+                CHECKED_REALLOC_NO_PTR(out_string, out_string_len, point_buf_size, H5E_DATASPACE, FAIL);
+
+                if (H5Sget_select_elem_pointlist(space_id, 0, (hsize_t)num_points, (hsize_t *)out_string) < 0)
```

Comment:
Have you tested point selections to verify that this approach works? Whereas for hyperslabs the selection is actually converted into a string, this appears to be just inserting `hsize_t` values into a `char *` buffer, which is not really what this function is aiming to do. It's also not correct to cast a `char *` (`out_string`) to `hsize_t *` here due to alignment requirements.

At the very least, if you're really just wanting a binary buffer of the `hsize_t` values for each point coordinate, that functionality should like be in a separate function to make it clear that we aren't actually expecting a string back.